test/code/plugins/VMInsightsDataCollector_subtest.rb

```diff
@@ -1127,27 +1130,29 @@ def initialize(dev, mp, size, free, type=(Random.rand(3) + 2))
             end
 
             def to_s
-                "%-10s %-15s %15d %15d" % [ @dev, @mp, @size, @free ]
+                "%-10s %-15s %15d %15d %-10s" % [ @dev, @mp, @size, @free, @type ]
             end
 
             def equivilent?(actual)
                 @dev == '/dev/' + actual.device_name &&
                 @mp == actual.mount_point &&
                 @size == actual.size_in_bytes &&
-                @free == actual.free_space_in_bytes
+                @free == actual.free_space_in_bytes &&
+                @type == actual.filesystem
             end
 
             def <=>(o)
                 r = @dev <=> o.dev; return r unless r.zero?
                 r = @mp <=> o.mp; return r unless r.zero?
                 r = @size <=> o.size; return r unless r.zero?
                 @free <=> o.dev
```

Comment:
What the??? This line, `@free <=> o.dev` is wrong! free should be compared to free, not dev!